include/simdjson/generic/ondemand/document-inl.h

```diff
@@ -68,7 +68,10 @@ simdjson_inline simdjson_result<object> document::start_or_resume_object() noexc
 simdjson_inline simdjson_result<value> document::get_value() noexcept {
   // Make sure we start any arrays or objects before returning, so that start_root_<object/array>()
   // gets called.
-  iter.assert_at_document_depth();
+#if SIMDJSON_DEVELOPMENT_CHECKS
+  if (!iter.at_root()) { return OUT_OF_ORDER_ITERATION; }
+#endif
+  iter.assert_at_root();
```

Comment:
BTW in our wrapper we always call `rewind()` when casting a document to a value, and it works even if the iteration was started.

Reply:
If you rewind the document, yes, it would work. I will add a test and documentation for that case.

Reply:
Test added and documentation improved.